RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitConvertPortable2CPUFormat.cc

```diff
@@ -0,0 +1,107 @@
+      uncalibRecHitsCPUEBToken_{
+          produces<EBUncalibratedRecHitCollection>(ps.getParameter<std::string>("recHitsLabelCPUEB"))},
+      uncalibRecHitsCPUEEToken_{
+          isPhase2_ ? edm::EDPutTokenT<EEUncalibratedRecHitCollection>{}
+                    : produces<EEUncalibratedRecHitCollection>(ps.getParameter<std::string>("recHitsLabelCPUEE"))} {}
+
+EcalUncalibRecHitConvertPortable2CPUFormat::~EcalUncalibRecHitConvertPortable2CPUFormat() {}
+
+void EcalUncalibRecHitConvertPortable2CPUFormat::produce(edm::Event &event, edm::EventSetup const &setup) {
+  auto const &uncalRecHitsEBColl = event.get(uncalibRecHitsPortableEB_);
+  auto const &uncalRecHitsEBCollView = uncalRecHitsEBColl.const_view();
+  auto recHitsCPUEB = std::make_unique<EBUncalibratedRecHitCollection>();
+  recHitsCPUEB->reserve(uncalRecHitsEBCollView.size());
+
+  for (uint32_t i = 0; i < uncalRecHitsEBCollView.size(); ++i) {
+    recHitsCPUEB->emplace_back(DetId{uncalRecHitsEBCollView.id()[i]},
+                               uncalRecHitsEBCollView.amplitude()[i],
+                               uncalRecHitsEBCollView.pedestal()[i],
+                               uncalRecHitsEBCollView.jitter()[i],
+                               uncalRecHitsEBCollView.chi2()[i],
+                               uncalRecHitsEBCollView.flags()[i]);
+    if (isPhase2_) {
+      (*recHitsCPUEB)[i].setAmplitudeError(uncalRecHitsEBCollView.amplitudeError()[i]);
+    }
+    (*recHitsCPUEB)[i].setJitterError(uncalRecHitsEBCollView.jitterError()[i]);
+    for (uint32_t sample = 0; sample < EcalDataFrame::MAXSAMPLES; ++sample) {
+      (*recHitsCPUEB)[i].setOutOfTimeAmplitude(sample, uncalRecHitsEBCollView.outOfTimeAmplitudes()[i][sample]);
+    }
+  }
+
+  if (!isPhase2_) {
+    auto const &uncalRecHitsEEColl = event.get(uncalibRecHitsPortableEE_);
+    auto const &uncalRecHitsEECollView = uncalRecHitsEEColl.const_view();
+    auto recHitsCPUEE = std::make_unique<EEUncalibratedRecHitCollection>();
+    recHitsCPUEE->reserve(uncalRecHitsEECollView.size());
+
+    for (uint32_t i = 0; i < uncalRecHitsEECollView.size(); ++i) {
+      recHitsCPUEE->emplace_back(DetId{uncalRecHitsEECollView.id()[i]},
+                                 uncalRecHitsEECollView.amplitude()[i],
+                                 uncalRecHitsEECollView.pedestal()[i],
+                                 uncalRecHitsEECollView.jitter()[i],
+                                 uncalRecHitsEECollView.chi2()[i],
+                                 uncalRecHitsEECollView.flags()[i]);
+      (*recHitsCPUEE)[i].setJitterError(uncalRecHitsEECollView.jitterError()[i]);
+      for (uint32_t sample = 0; sample < EcalDataFrame::MAXSAMPLES; ++sample) {
+        (*recHitsCPUEE)[i].setOutOfTimeAmplitude(sample, uncalRecHitsEECollView.outOfTimeAmplitudes()[i][sample]);
+      }
+    }
+    event.put(uncalibRecHitsCPUEEToken_, std::move(recHitsCPUEE));
+  }
+  event.put(uncalibRecHitsCPUEBToken_, std::move(recHitsCPUEB));
```

Comment:
Just for clarity, could you move this line above, before the `if (!isPhase2_) { ...` line ?